packages/components/src/context-tag/context-tag.tsx

```diff
@@ -1,93 +1,230 @@
+      }
+      case 'audio': {
+        return (
+          <>
+            <Avatar
+              type="icon"
+              size={avatarSizes[size]}
+              icon={<PlayIcon size={16} />}
+              backgroundColor="$primary-50"
+              color="$white-100"
+            />
+            <Label size={size}>{props.audioLength}</Label>
+          </>
+        )
+      }
+      case 'account': {
+        return (
+          <>
+            <Avatar
+              type="account"
+              size={avatarSizes[size]}
+              name={props.user.emoji}
+            />
+            <Label size={size}>{props.user.name}</Label>
+          </>
+        )
+      }
+      case 'group': {
+        return (
+          <>
+            <Avatar
+              type="icon"
+              size={avatarSizes[size]}
+              backgroundColor="$purple-50"
+              color="$white-70"
+              icon={props.group.icon}
+            />
+            <Label size={size}>{props.group.name}</Label>
+          </>
+        )
+      }
+      case 'network': {
+        return (
+          <>
+            <Avatar
+              type="community"
+              size={avatarSizes[size]}
+              src={props.network.src}
+              name={props.network.name}
+            />
+            <Label size={size}>{props.network.name}</Label>
+          </>
+        )
+      }
+      case 'collectible': {
+        return (
+          <>
+            <Avatar
+              type="account"
+              size={avatarSizes[size]}
+              src={props.collectible.src}
+              name={props.collectible.name}
+            />
+            <Label size={size}>{props.collectible.name}</Label>
+          </>
+        )
+      }
+      case 'icon': {
+        return (
+          <>
+            <Avatar
+              type="icon"
+              size={avatarSizes[size]}
+              icon={props.icon}
+              backgroundColor="$transparent"
+              color="$neutral-50"
+            />
+            <Label size={size}>{props.label}</Label>
+          </>
+        )
+      }
+    }
+  }
 
-      {Array.isArray(label) ? (
-        label.map((item, i) => {
-          if (i !== 0) {
-            return (
-              <Fragment key={item}>
-                <ChevronRightIcon size={16} color="$neutral-50" />
-                <Label size={size}>{item}</Label>
-              </Fragment>
-            )
-          } else {
-            return (
-              <Label size={size} key={item}>
-                {item}
-              </Label>
-            )
-          }
-        })
-      ) : (
-        <Label size={size}>{label}</Label>
-      )}
+  return (
+    <Base
+      outline={outline}
+      blur={blur}
+      borderRadius={rounded ? '$8' : '$full'}
+      size={size}
+      gap={size === 24 ? 4 : 8}
+      paddingVertical={leftShelf ? 0 : 1}
+      paddingHorizontal={size === 24 ? 8 : 12}
+      paddingLeft={leftShelf ? 1 : size === 24 ? 8 : 12}
```

Comment:
Unless I am missing something, I would move `size` specific styling into variants instead of using ternary operator inline – I think it's more readable, because styles for each variant are colocated. This only applies to `gap` and `paddingHorizontal`.